sklearn/discriminant_analysis.py

```diff
@@ -120,11 +120,11 @@ def _class_cov(X, y, priors=None, shrinkage=None):
         Class covariance matrix.
     """
     classes = np.unique(y)
-    covs = []
-    for group in classes:
+    cov = np.zeros(shape=(X.shape[1], X.shape[1]))
+    for idx, group in enumerate(classes):
         Xg = X[y == group, :]
-        covs.append(np.atleast_2d(_cov(Xg, shrinkage)))
-    return np.average(covs, axis=0, weights=priors)
+        cov += priors[idx] * np.atleast_2d(_cov(Xg, shrinkage))
```

Comment:
This line does not work if `priors is None`. You should do something like:
```py
cov_g = np.atleast_2d(_cov(Xg, shrinkage))
if priors is not None:
    cov_g *= priors[idx]
cov += cov_g
```

Reply:
Oh priors cannot be None.
Then maybe we should remove the default which implies `None` is a valid input.